hack/verify-static-check.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+GOLANGCI_LINT_PKG="github.com/golangci/golangci-lint/cmd/golangci-lint"
+GOLANGCI_LINT_VER="v1.46.2"
+
+cd "${REPO_ROOT}"
+source "hack/util.sh"
+
+util::install_tools ${GOLANGCI_LINT_PKG} ${GOLANGCI_LINT_VER}
+
+if golangci-lint run; then
+  echo 'Congratulations!  All Go source files have passed staticcheck.'
+else
+  echo # print one empty line, separate from warning messages.
+  echo 'Please review the above warnings.'
+  echo 'If the above warnings do not make sense, feel free to file an issue.'
+  exit 1
+fi
```

Comment:
The logic within `make lint` is duplicated in this script, so we can choose one of them.

Reply:
done